internal/cmd/plugin/report/olm.go

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright The CloudNativePG Contributors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package report
+
+import (
+	"archive/zip"
+	"context"
+	"fmt"
+	"path/filepath"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/cloudnative-pg/cloudnative-pg/internal/cmd/plugin"
+)
+
+// getOlmResource gets the desired resource using the Dynamic Client
+// to avoid code dependencies on the OLM libraries
+func getOlmResource(
+	ctx context.Context,
+	dynamicClient dynamic.Interface,
+	namespace, resource string,
+) (client.ObjectList, error) {
+	isContained := func(item unstructured.Unstructured, list []unstructured.Unstructured) bool {
+		for _, element := range list {
+			if element.GetName() == item.GetName() {
+				return true
+			}
+		}
+		return false
+	}
+
+	gvr := schema.GroupVersionResource{
+		Group:    "operators.coreos.com",
+		Version:  "v1alpha1",
+		Resource: resource,
+	}
+
+	listFromLabelOpenshift, err := dynamicClient.Resource(gvr).Namespace(namespace).
+		List(ctx, metav1.ListOptions{LabelSelector: labelOpenshiftOperators})
+	if err != nil {
+		return nil, fmt.Errorf("could note get resource: %v, %v", gvr, err)
+	}
+
+	listFromLabelNamespace, err := dynamicClient.Resource(gvr).Namespace(namespace).
+		List(ctx, metav1.ListOptions{LabelSelector: getLabelOperatorsNamespace()})
+	if err != nil {
+		return nil, fmt.Errorf("could note get resource: %v, %v", gvr, err)
+	}
+
+	for idx := range listFromLabelOpenshift.Items {
+		item := listFromLabelOpenshift.Items[idx]
+		if !isContained(item, listFromLabelNamespace.Items) {
+			listFromLabelOpenshift.Items = append(listFromLabelOpenshift.Items, item)
+		}
+	}
+
+	return listFromLabelOpenshift, err
```

Comment:
This won't merge the lists. If the operator isn't in `openshift-operators`, but we have an operator in a namespace, we return empty list.